test/testutil.h

```diff
@@ -291,8 +291,8 @@ void test_info_c90(const char *desc, ...) PRINTF_FORMAT(1, 2);
 # define TEST_mem_eq(a, m, b, n) test_mem_eq(__FILE__, __LINE__, #a, #b, a, m, b, n)
 # define TEST_mem_ne(a, m, b, n) test_mem_ne(__FILE__, __LINE__, #a, #b, a, m, b, n)
 
-# define TEST_true(a)         test_true(__FILE__, __LINE__, #a, a)
-# define TEST_false(a)        test_false(__FILE__, __LINE__, #a, a)
+# define TEST_true(a)         test_true(__FILE__, __LINE__, #a, (a) != 0)
+# define TEST_false(a)        test_false(__FILE__, __LINE__, #a, (a) == 0)
```

Comment:
Unless I gravely misunderstand, both of these test that `(a) != 0`...  that doesn't seem quite right